File: backend/src/main/java/com/traveloop/service/interfaces/CityService.java
package com.traveloop.service.interfaces;

import com.traveloop.model.dto.response.CityResponse;

import java.util.List;
import java.util.UUID;

public interface CityService {

    List<CityResponse> getAllCities();

    CityResponse getCityById(UUID cityId);

    List<CityResponse> searchCities(String query);

    List<CityResponse> getPopularCities();

    List<CityResponse> getCitiesByCountry(String country);

    List<CityResponse> getBudgetFriendlyCities(Integer maxCostIndex);

    List<String> getAllCountries();
}

File: frontend/src/services/cityService.js
/* ────────────────────────────────────────────
   cityService — City/Destination API calls
   ──────────────────────────────────────────── */

import api from './api';

const cityService = {
  getAllCities: () => api.get('/api/cities').then((r) => r.data),

  getCityById: (cityId) =>
    api.get(`/api/cities/${cityId}`).then((r) => r.data),

  searchCities: (query) =>
    api.get('/api/cities/search', { params: { q: query } }).then((r) => r.data),

  getPopularCities: () => api.get('/api/cities/popular').then((r) => r.data),

  getCitiesByCountry: (country) =>
    api.get(`/api/cities/country/${country}`).then((r) => r.data),

  getBudgetCities: (maxCost) =>
    api.get('/api/cities/budget', { params: { maxCost } }).then((r) => r.data),

  getAllCountries: () => api.get('/api/cities/countries').then((r) => r.data),
};

export default cityService;
